File: recursion/javascript/sumArray.js
// sum all values in an array

function sumArrayIterative(arr) {
    let total = 0;

    for (val of arr) {
        total += val;
    }

    return total;
}

function sumArrayRecursive(arr) {
    if (arr.length === 0) {
        return 0;
    } else {
        const head = arr[0];
        const tail = arr.slice(1, arr.length);
        return head + sumArrayRecursive(tail);
    }
}

console.log(sumArrayIterative([1, 2, 3, 4, 5]));
console.log(sumArrayRecursive([1, 2, 3, 4, 5]));

File: recursion/python/sum_array.py
def sum_array(arr):
    if len(arr) == 0:
        return 0
    else:
        head = arr[0]
        tail = arr[1:]
        return head + sum_array(tail)


if __name__ == '__main__':
    print(sum_array([1, 2, 3, 4, 5]))
